sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ErrorParserSuite.scala

```diff
@@ -63,4 +63,78 @@ class ErrorParserSuite extends SparkFunSuite {
       "Combination of ORDER BY/SORT BY/DISTRIBUTE BY/CLUSTER BY is not supported",
       "^^^")
   }
+
+  test("hyphen in identifier - DDL tests") {
+    val msg = "unquoted identifier"
+    // ddl tests
+    intercept("use test-test", 1, 8, 9, msg + " test-test", "^^^")
+    intercept("create database if not exists my-database", 1, 32, 33, "my-database", "^^^")
+    intercept("DROP DATABASE my-database", 1, 16, 17, "my-database", "^^^")
+    intercept(
+      """
+        |ALTER DATABASE my-database
+        |SET DBPROPERTIES ('p1'='v1')""".stripMargin, 2, 17, 18, "my-database", "^^^")
+    intercept("CREATE TABLE test (attri-bute INT)", 1, 24, 25, msg + " attri-bute", "^^^")
+    intercept("ALTER TABLE test ADD COLUMNS (h-col BIGINT)", 1, 31, 32, msg + " h-col", "^^^")
+    intercept(
+      """
+        |CREATE TABLE IF NOT EXISTS mydb.page-view
+        |USING parquet
+        |COMMENT 'This is the staging page view table'
+        |LOCATION '/user/external/page_view'
+        |TBLPROPERTIES ('p1'='v1', 'p2'='v2')
+        |AS SELECT * FROM src""".stripMargin, 2, 36, 37, msg + " page-view", "^^^")
+    intercept("analyze table test-table PARTITION (part1)", 1, 18, 19, msg + " test-table", "^^^")
+    intercept("analyze table test PARTITION (part-i)", 1, 34, 35, msg + " part-i", "^^^")
+    intercept("drop function my-func if exists", 1, 16, 17, " my-func", "^^^")
+    intercept("show tables in hyphen-database", 1, 21, 22, " hyphen-database", "^^^")
+    intercept("show test-id functions like \"str\" ", 1, 9, 10, " test-id", "^^^")
+    intercept("load data inpath \"path\" into table my-tab", 1, 37, 38, " my-tab", "^^^")
+
+  }
+
+  test("hyphen in identifier - DML tests") {
+    val msg = "unquoted identifier"
+    // dml tests
+    intercept("SELECT * FROM table-with-hyphen", 1, 19, 25, msg + " table-with-hyphen", "^^^")
+    // special test case: minus in expression shouldn't be treated as hyphen in identifiers
+    intercept("SELECT a-b FROM table-with-hyphen", 1, 21, 27, msg + " table-with-hyphen", "^^^")
+    intercept("SELECT a-b FROM table-hyphen WHERE a-b = 0", 1, 21, 22, msg + " table-hyphen", "^^^")
+    intercept("SELECT (a - test_func(b-c)) FROM test-table", 1, 37, 38, msg + " test-table", "^^^")
+    intercept("WITH a-b AS (SELECT 1 FROM s) SELECT * FROM s;", 1, 6, 7, msg + " a-b", "^^^")
+    intercept(
+      """
+        |SELECT a, b
+        |FROM grammar-breaker
+        |WHERE a-b > 10
+        |GROUP BY fake-breaker
```

Comment:
Actually, this is hidden by `grammar-breaker`. We had better have a positive case to make it sure `fake-breaker` will work to be complete.